src/aws/osml/models/sm_logger.py

```diff
@@ -0,0 +1,131 @@
+#  Copyright 2023-2024 Amazon.com, Inc. or its affiliates.
+
+import json
+import logging
+import time
+from typing import Optional
+
+import boto3
+
+from aws.osml.models.app_config import BotoConfig
+
+
+class SMLogger(logging.Handler):
```

Comment:
I don't think we need to do most of this. 

SageMaker should already be transferring everything from stdout and stderr to standardized log group names for endpoints. group = /aws/sagemaker/Endpoints/[EndpointName] stream = [production-variant-name]/[instance-id]. We shouldn't move the log records to a non-standard place.

We should be able to setup a JSON formatter and log context on the handler writing to stdout and get everything sent to CloudWatch by SageMaker.

See: https://docs.aws.amazon.com/sagemaker/latest/dg/logging-cloudwatch.html